src/test/java/graphql/testrunner/service/CommandExecutorServiceTest.java

```diff
@@ -0,0 +1,222 @@
+package graphql.testrunner.service;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import graphql.testrunner.util.TestRunnerException;
+
+import static java.util.Arrays.asList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isA;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CommandExecutorServiceTest {
+
+    @Mock
+    private ProcessBuilder processBuilder;
+
+    @Mock
+    private static Logger LOGGER;
+
+    @Mock
+    private Process process;
+
+    @Mock
+    private static BufferedReader bufferedReader;
+
+    @InjectMocks
+    private CommandExecutorService commandExecutorService = new CommandExecutorService(){
+        @Override
+        ProcessBuilder getBuilder(String dir) {
+            return processBuilder;
+        }
+        @Override
+        BufferedReader getReader(Process p) {
+            return bufferedReader;
+        }
+
+    };
+
+    @Test
+    void executeCommand_when_exit_code_is_zero() throws Exception {
+        List<String> command = asList("SH");
+        setupProcessBuilderMock(command, 0);
+
+        commandExecutorService.executeCommand(command);
+        InOrder inOrder = inOrder(LOGGER);
+        inOrder.verify(LOGGER).log(eq(Level.INFO), eq("Executing command : {0}"),
+          eq(command));
+        String dir = null;
+        inOrder.verify(LOGGER).log(eq(Level.INFO), eq("In path : {0}"), eq(dir));
```

Comment:
in the future, no need to verify that the logger is doing what you want, it doesn't hurt, but that is usually where I draw the line in terms of effort